packages/amd-loader/amd-loader.js

```diff
@@ -0,0 +1,196 @@
+'use strict';
+
+(function() {
+
+function Registry() {
+  this.modules = {};
+  this.topLevelIdx = 0;
+  this.previousTopLevelUrl = undefined;
+  this.pendingDefine = undefined;
```

Comment:
Document these fields please. Maybe use `// @ts-check` and jsdoc to get basic type checking, or use full typescript and tsc can also handle transpiling to ES5

Reply:
+1 on TypeScript